### PR TITLE
Updated project structure

### PR DESCRIPTION
**Pull request by devthedevel**
**Context**
Updated project structure. This includes some quality improvements (such as a .gitignore file), and added Serverless framework.

### CODE REVIEW

File: src/index.js [hunk 1]
```diff
@@ -4,12 +4,15 @@
...
+    console.log('Validating event...')
+
+    console.log('Event validated!')
     // Dummy function for design
     return{}
 }
 
 
-export const handler = async (event) => {
+module.exports.handler = async (event) => {
```
Comment (devthedevel) [inline src/index.js:15]: Had to revert to non ES6 syntax. I couldn't get VS code to work with it. I will keep looking into it though.

File: __tests__/data/sample_request_1.json [hunk 2]
```diff
@@ -1,3 +1,9 @@
...
-    "body": "{}"
+{ 
+    "method": "GET",
+    "body": { },
+    "query": {
+        "race": "ELF"
+    },
+    "path": { },
+    "requestPath": "/v1/generate/character"
 }
```
Comment (devthedevel) [inline __tests__/data/sample_request_1.json]: This is more like what an actual API Gateway request would be like (without the headers)

**Review by krg276 — COMMENTED** [review]
I checked everything over, is the serverless.yml just a way to define the AWS lambda functions?

**Comment by devthedevel** [comment]
> 
> 
> I checked everything over, is the serverless.yml just a way to define the AWS lambda functions?

Yeah the serverless.yml is part of the serverless framework. It helps setup, configure, and deploy to AWS services. I added a AWS API Gateway to the function, so you can invoke it using HTTP.

**Pull request closed without merging**